finetuner/tuner/callback/model_checkpoint.py

```diff
@@ -0,0 +1,134 @@
+from typing import TYPE_CHECKING, Optional
+from .base import BaseCallback
+import os
+import numpy as np
+from finetuner.helper import get_framework
+from jina.logging.logger import JinaLogger
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class ModelCheckpointCallback(BaseCallback):
+    """
+    Callback to save model at some frequency (epochs or batchs)
```

Comment:
this docstring does not seem accurate, `batches` do not seem to be considered at all.